source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -70,6 +70,8 @@ DnsCacheImpl::DnsCacheImpl(
   }
   enable_dfp_dns_trace_ = context.serverFactoryContext().runtime().snapshot().getBoolean(
       "envoy.enable_dfp_dns_trace", false);
+  enable_dfp_resolve_timeout_ = context.serverFactoryContext().runtime().snapshot().getBoolean(
+      "envoy.enable_dfp_resolve_timeout", true);
```

Comment:
Do you think anyone else might want this feature? If so we should probably add it to DFP config

Reply:
I updated the PR to use the existing `DnsCacheConfig.dns_query_timeout` by setting it to 0 to disable the DNS query timeout.